single_kernel_mongo/managers/mongodb_operator.py

```diff
@@ -277,7 +289,7 @@ def on_start(self) -> None:
             return
 
         self._initialise_replica_set()
-        self.charm.status_manager.to_active(None)
+        self.charm.status_manager.process_and_share_statuses()
```

Comment:
Just a heads up you still have several calls to `process_and_share_statuses`, which can be troublesome, see earlier comment of:

> `process_and_share_statuses()` doesn't contain all the feasible statuses for mongodb so it has the possibility to overwrite a status that was set earlier (this will be fixed when we implement statuses) 

So having this call at the end of each hook can be problematic